### PR TITLE
feat(kayenta): pass the accountId to Kayenta for deployments

### PR DESCRIPTION
**Pull request by skandragon**
Things that did not change:

* For non-deployments, all scope data is passed through unchanged.

Changes:
* Attempt to retrieve the accountId from Mort for the deployment we have just made, if any.
* If it is present, set it in the `extendedScopeParams` under the key `accountId` for both control and experiment.
* Stop doing a very late override in the stage that runs the canary report; instead, this stage will now pass along exactly the scopes provided in its context.
* This pushes all the heavy lifting of "is this a deployment or not" as well as temporal planning to the `RunCanaryIntervalsStage`.

### CODE REVIEW

**Comment by skandragon** [comment]
@robfletcher, I added you as this is Kotlin and you wrote the original.  Please comment on anything, but especially the areas where I could Kotlinize things better.

@dufter, please pay special attention to the accountId fetching from Mort and its subsequent handling.

File: orca-kayenta/src/main/kotlin/com/netflix/spinnaker/orca/kayenta/pipeline/RunCanaryIntervalsStage.kt [hunk 1]
```diff
@@ -46,6 +47,18 @@ class RunCanaryIntervalsStage(private val clock: Clock) : StageDefinitionBuilder
   override fun taskGraph(stage: Stage, builder: TaskNode.Builder) {
   }
 
+  private fun getDeployDetails(stage: Stage) : DeployedServerGroupContext? {
+    val deployedServerGroupsStage = stage.parent?.execution?.stages?.find {
```
Comment (ajordens) [inline orca-kayenta/src/main/kotlin/com/netflix/spinnaker/orca/kayenta/pipeline/RunCanaryIntervalsStage.kt]: IMO, you'll want to use `StageNavigator` and not just look for the first matching stage in the execution.

The navigator will look up the graph and allow you to grab the "closest" stage.

Comment (skandragon) [reply orca-kayenta/src/main/kotlin/com/netflix/spinnaker/orca/kayenta/pipeline/RunCanaryIntervalsStage.kt]: I'll look into this, thanks

Comment (ajordens) [reply orca-kayenta/src/main/kotlin/com/netflix/spinnaker/orca/kayenta/pipeline/RunCanaryIntervalsStage.kt]: Scratch that comment ... misunderstanding around the shape of this graph.

**Pull request merged**